src/network/event_loop.rs

```diff
@@ -194,12 +194,18 @@ impl EventLoop {
 						QueryResult::PutRecord(result) => {
 							if let Some(v) = self.pending_kad_query_batch.get_mut(&id) {
 								*v = Some(result.map_err(Into::into).map(|ok_res| {
+									// Remove local records for fat clients (memory optimization)
 									if self.kad_remove_local_record {
 										self.swarm
 											.behaviour_mut()
 											.kademlia
 											.remove_record(&ok_res.key);
 									};
+									self.swarm
```

Comment:
always shrinking on each put? without any checks?

Reply:
not ideal, a better solution would be to shrink it on every block, but the call itself is very resource cheap even with this approach - I'm seeing no performance penalties under a load test...

Reply:
Maybe its easy to invoke it on batch complete?